pallets/order-book/src/lib.rs

```diff
@@ -665,12 +696,35 @@ pub mod pallet {
 				currency_in: order.asset_in_id,
 				currency_out: order.asset_out_id,
 				fulfillment_amount: buy_amount,
-				sell_rate_limit: order.max_sell_rate,
+				max_sell_price: order.max_sell_price,
 			});
 
 			Ok(())
 		}
 
+		pub fn market_price(
+			currency_in: T::AssetCurrencyId,
+			currency_out: T::AssetCurrencyId,
+		) -> Result<T::SellRatio, DispatchError> {
+			let price = T::PriceProvider::get(
+				&T::MarketFeederId::get(),
+				&(currency_in, currency_out).into(),
+			)?;
+
+			Ok(match price {
+				Some(price) => price,
+				None => {
+					let price = T::PriceProvider::get(
+						&T::MarketFeederId::get(),
+						&(currency_out, currency_in).into(),
+					)?
+					.ok_or(Error::<T>::MarketPriceNotFound)?;
+
+					T::SellRatio::one().ensure_div(price)?
+				}
+			})
+		}
```

Comment:
This is the key to getting it to work with just one method, conserving the `place_order()` API.

Reply:
Depending on how you specify the currency parameters, you'll obtain the market price or the inverse one